### PR TITLE
Scrum 8 post task

### PR DESCRIPTION
**Pull request by FreddyFZ17**
Added all smoke's testcases to the delete endpoint, fixed some issues and improved zetup and teardown implementations.

### CODE REVIEW

**Review by Jasonn5 — APPROVED** [review]
Executed succesfully, good job freddy

File: api_endpoints/api_request.py
Comment (AlisonGuzman2006) [inline api_endpoints/api_request.py:1]: The changes has been reviewed and accepted

File: tests/task/conftest.py
Comment (AlisonGuzman2006) [inline tests/task/conftest.py:1]: The changes has been reviewed and accepted

File: tests/task/test_delete_task.py
Comment (AlisonGuzman2006) [inline tests/task/test_delete_task.py:1]: The addition of the code has also been reviewed! Great job, all works correctly!

**Review by AlisonGuzman2006 — APPROVED** [review]
The changes and all addition of the code have been reviewed! Great job, all works correctly!

**Pull request merged**